src/sinks/util/batch.rs

```diff
@@ -109,12 +100,18 @@ where
     ///
     /// This is based on the uncompressed size of the batched events, before they are
     /// serialized / compressed.
+    #[serde(default = "default_max_bytes::<D>")]
+    #[configurable(metadata(docs::type_unit = "bytes"))]
     pub max_bytes: Option<usize>,
 
-    /// The maximum size of a batch, in events, before it is flushed.
+    /// The maximum size of a batch before it is flushed.
+    #[serde(default = "default_max_events::<D>")]
+    #[configurable(metadata(docs::type_unit = "events"))]
     pub max_events: Option<usize>,
 
-    /// The maximum age of a batch, in seconds, before it is flushed.
+    /// The maximum age of a batch before it is flushed.
+    #[serde(default = "default_timeout::<D>")]
+    #[configurable(metadata(docs::type_unit = "seconds"))]
     pub timeout_secs: Option<f64>,
```

Comment:
We've been generally making fields that represent time a Duration like this: https://github.com/vectordotdev/vector/blob/master/src/sources/gcp_pubsub.rs#L159-L161

Is that possible here?

Reply:
The configurable attribute doesn't properly handle the `Option` wrapped `Durations`

Reply:
^  as Spencer indicated.
I did give it a try, even though it didn't work on one of the sources I tried it on earlier.